storage/innobase/include/fil0crypt.h

```diff
@@ -327,6 +327,46 @@ struct fil_space_scrub_status_t {
   ulint current_scrub_max_page_number; /*!< current scrub max page no */
 };
 
+/** Handles the fetching/generation/storing/etc of keyring redo log keys.
+
+ This class is *NOT* thread safe, as thread safety is not required.
+ Data is only accessed/modified on the following points:
+ * When the redo space is created, at startup
+ * During redo log recovery, at startup
+ * When the server UUID is generated, at startup
+ * When the user requests a new key version, checked periodically in the
+   master thread
+
+ As these can't happen in parallel, no lock is used.
+ */
+class redo_log_keys {
+ public:
+  MY_NODISCARD
+  redo_log_key *load_latest_key();
+  MY_NODISCARD
+  redo_log_key *load_key_version(uint version);
+
+  MY_NODISCARD
+  redo_log_key *generate_and_store_new_key();
+
+  /** These two methods are used during bootstrap encryption,
+    when wo do not yet have an uuid */
+  MY_NODISCARD
+  redo_log_key *generate_new_key_without_storing();
+
+  MY_NODISCARD
+  bool store_used_keys();
+
+  void unload_old_keys();
+
+ private:
+  typedef std::map<ulint, redo_log_key> key_map;
+  typedef key_map::iterator key_iterator;
```

Comment:
const_iterator would make sense too. Perhaps, if auto is used, then there is no need for this typedef at all